shared/src/main/scala/org/yaml/render/ScalarRender.scala

```diff
@@ -54,7 +54,7 @@ object ScalarRender {
     if (l == 0)
       return if (mustBeString || !mark.plain) DoubleQuoteMark else NoMark
 
-    if (text.head == ' ' || text.endsWith("\n\n")) return DoubleQuoteMark
+    if (text.head == ' ' || text.endsWith("\n\n") || text == "-") return DoubleQuoteMark
```

Comment:
Please, extract to method like "shouldQuoteText" or "requireDobleQuote"

Reply:
there are many things that require double quote in the code. Adding it just for 3 cases might be confusing for the reader